include/nbl/video/IPhysicalDevice.h

```diff
@@ -14,14 +16,57 @@ namespace video
 class IPhysicalDevice : public core::IReferenceCounted
 {
 public:
-    struct SProperties
+    struct SLimits
     {
-        // TODO
+        uint32_t UBOAlignment;
+        uint32_t SSBOAlignment;
+        uint32_t bufferViewAlignment;
+
+        uint32_t maxUBOSize;
+        uint32_t maxSSBOSize;
+        uint32_t maxBufferViewSizeTexels;
+        uint32_t maxBufferSize;
+
+        uint32_t maxPerStageSSBOs;
+        //uint32_t maxPerStageUBOs;
+        //uint32_t maxPerStageTextures;
+        //uint32_t maxPerStageStorageImages;
+
+        uint32_t maxSSBOs;
+        uint32_t maxUBOs;
+        uint32_t maxTextures;
+        uint32_t maxStorageImages;
+
+        float pointSizeRange[2];
+        float lineWidthRange[2];
+
+        uint32_t maxViewports;
+        uint32_t maxViewportDims[2];
+
+        uint32_t maxWorkgroupSize[3];
+
+        uint32_t subgroupSize;
+        std::underlying_type_t<asset::ISpecializedShader::E_SHADER_STAGE> subgroupOpsShaderStages;
     };
 
     struct SFeatures
     {
-        // TODO
+        bool robustBufferAccess = false;
+        bool imageCubeArray = false;
+        bool logicOp = false;
+        bool multiDrawIndirect = false;
+        bool multiViewport = false;
+        bool shaderSubgroupBasic = false;
+        bool shaderSubgroupVote = false;
+        bool shaderSubgroupArithmetic = false;
+        bool shaderSubgroupBallot = false;
+        bool shaderSubgroupShuffle = false;
+        bool shaderSubgroupShuffleRelative = false;
+        bool shaderSubgroupClustered = false;
+        bool shaderSubgroupQuad = false;
+        // Whether `shaderSubgroupQuad` flag refer to all stages where subgroup ops are reported to be supported.
+        // See SLimit::subgroupOpsShaderStages.
+        bool shaderSubgroupQuadAllStages = false;
```

Comment:
https://vulkan.gpuinfo.org/listpropertiescore10.php
Properties
https://vulkan.gpuinfo.org/listpropertiescore11.php
https://vulkan.gpuinfo.org/listpropertiescore12.php
https://vulkan.gpuinfo.org/listpropertiesextensions.php

Features
https://vulkan.gpuinfo.org/listfeaturescore10.php
https://vulkan.gpuinfo.org/listfeaturescore11.php
https://vulkan.gpuinfo.org/listfeaturescore12.php
https://vulkan.gpuinfo.org/listfeaturesextensions.php

Extensions
https://vulkan.gpuinfo.org/listextensions.php

Instance Extensions
https://vulkan.gpuinfo.org/listinstanceextensions.php

Formats
https://vulkan.gpuinfo.org/listformats.php

Memory Heaps
https://vulkan.gpuinfo.org/listmemory.php
(on GL you just pretend there's 3-4 heaps of unknown size)

Surface Formats and Present Modes (should something else report this?)
https://vulkan.gpuinfo.org/listsurfaceformats.php

Naturally you can collapse like-wise extensions into one feature
https://vulkan.gpuinfo.org/listsurfacepresentmodes.php

Reply:
@achalpandeyy